lua/treesitter-context.lua

```diff
@@ -109,14 +128,33 @@ local is_valid = cache.memoize(function(node, query)
       end
 
       local name = query.captures[id] -- name of the capture in the query
-      if not r and name == 'context' then
-        r = node == node0
-      elseif name == 'context.final' then
+      if name == 'context.final' then
         range[3] = erow
         range[4] = ecol
       elseif name == 'context.end' then
         range[3] = srow
         range[4] = scol
+      elseif not r and name == 'context' then
+        r = node == node0
+      elseif not r then
+        local category = name:match('^context%.(.*)$') 
+        local ft_categories = config.categories[vim.bo.filetype]
+        local ft_category = ft_categories and ft_categories[category]
+        if ft_category then
+          r = node == node0
+        elseif ft_category == nil then
+          local def_category = config.categories.default[category]
+          if def_category then
+            r = node == node0
+          elseif def_category == nil then
+            local lang = parsers.ft_to_lang(vim.bo.filetype)
```

Comment:
This only works in nvim 0.9.

Reply:
The same function is used in `get_parent_matches`. I'm also pretty sure it worked for me on `0.8.3`.

Reply:
The blame shows the function signature was last touched 3 years ago, so that should be fine.

Reply:
Oh this is because it's coming from nvim-treesitter, my bad.